### PR TITLE
[PTX-25143]ResizePoolDrivesInDifferentSize testcase fix

### PR DESCRIPTION
**Pull request by sayalasomayajula-px**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #
https://purestorage.atlassian.net/browse/PTX-25143

**Special notes for your reviewer**:
PASS LOGS: https://jenkins.pwx.dev.purestorage.com/job/Autopilot/job/tp-aut-nextpx-pks-pvc-and-pool-expand/684/display/redirect

### CODE REVIEW

File: tests/basic/storage_pool_test.go [hunk 1]
```diff
@@ -7699,7 +7706,7 @@ var _ = Describe("{NodeAddDiskWhileAddDiskInProgress}", func() {
 	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/51356
 	var runID int
 	JustBeforeEach(func() {
-		StartTorpedoTest("NodeAddDriveWhileAddDriveInProgress", "Initiate pool expansion using add-drive while one already in progress", nil, testrailID)
+		StartTorpedoTest("NodeAddDriveWhileAddDiskInProgress", "Initiate pool expansion using add-drive while one already in progress", nil, testrailID)
```
Comment (vumesh-px) [inline tests/basic/storage_pool_test.go:7709]:
```suggestion
		StartTorpedoTest("NodeAddDiskWhileAddDiskInProgress", "Initiate pool expansion using add-drive while one already in progress", nil, testrailID)
```

**Review by vumesh-px — CHANGES_REQUESTED** [review]

File: tests/basic/storage_pool_test.go [hunk 2]
```diff
@@ -5657,19 +5667,20 @@ func addDiskToSpecificPool(node node.Node, sizeOfDisk uint64, poolID int32) bool
 	newSpec := strings.Join(paramsArr, ",")
 	log.InfoD("New Spec Details %v", newSpec)
 
+	// Add Drive to the Volume
+	err = Inst().V.AddCloudDrive(&node, newSpec, poolID)
```
Comment (lsrinivas-pure) [inline tests/basic/storage_pool_test.go:5671]: i think this is repeated, please check

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

File: tests/basic/pool_function_test.go [hunk 3]
```diff
@@ -473,7 +473,7 @@ var _ = Describe("{PoolExpandDiskResizePXRestart}", func() {
 	})
 
 	It("Restart PX after pool expansion", func() {
-		StartTorpedoTest("RestartAfterPoolExpansion",
+		StartTorpedoTest("PoolExpandDiskResizePXRestart",
```
Comment (kphalgun-px) [inline tests/basic/pool_function_test.go:476]: 👍

File: tests/basic/storage_pool_test.go [hunk 4]
```diff
@@ -5737,13 +5744,13 @@ var _ = Describe("{ResizePoolDrivesInDifferentSize}", func() {
 		log.FailOnError(err, "Getting NodeID from the given poolUUID [%v] Failed", poolUUID)
 		log.InfoD("Node Details %v", nodeDetails)
 
-		log.InfoD("Adding New Disk with Size [%v]", diskSize)
-		response := addDiskToSpecificPool(*nodeDetails, diskSize, poolID)
+		log.InfoD("Adding New Disk with Size [%v] which is greater than the pool size available", diskSize)
+		response, _ := addDiskToSpecificPool(*nodeDetails, diskSize, poolID)
```
Comment (kphalgun-px) [inline tests/basic/storage_pool_test.go:5748]: What about the error that addDiskToSpecificPool returns?

File: tests/basic/storage_pool_test.go [hunk 5]
```diff
@@ -5704,23 +5717,17 @@ var _ = Describe("{ResizePoolDrivesInDifferentSize}", func() {
...
 		allPools, _ := Inst().V.ListStoragePools(metav1.LabelSelector{})
 		log.InfoD("List of all the Pools present in the system [%s]", allPools)
+		for _, pool := range allPools {
+			if pool.TotalSize > maxPoolSize {
+				maxPoolSize = pool.TotalSize / units.GiB
+			}
+		}
+		// Taking disksize 2 times the pool size to test negative scenario(to add disk size which is more than pool size)
+		diskSize := maxPoolSize * 2
+		poolUUID := pickPoolToResize(contexts, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, diskSize)
```
Comment (kphalgun-px) [inline tests/basic/storage_pool_test.go:5729]: We might need a check if poolUUID is empty.

File: tests/basic/storage_pool_test.go [hunk 6]
```diff
@@ -5628,25 +5628,35 @@ var _ = Describe("{PoolIncreaseSize20TB}", func() {
...
+				if err != nil {
+					return false, fmt.Errorf("Error getting pool details on node %s", node.Name)
+				}
+				for _, pool := range poolDetails {
+					if pool.ID == poolID {
+						poolSize = pool.TotalSize / units.GiB
+					}
+					specSize = poolSize
+				}
+				paramsArr = append(paramsArr, fmt.Sprintf("size=%d", specSize))
```
Comment (kphalgun-px) [inline tests/basic/storage_pool_test.go:5659]:
```suggestion
				for _, pool := range poolDetails {
					if pool.ID == poolID {
						poolSize := pool.TotalSize / units.GiB
						paramsArr = append(paramsArr, fmt.Sprintf("size=%d", poolSize))
					}
				}
```

?

**Review by vumesh-px — APPROVED** [review]

**Pull request merged**